### PR TITLE
[back] Fix X-TAXII-Date-Added-First/Last headers

### PR DESCRIPTION
**Pull request by rlynch-ironnet**
### Proposed changes

* Fix how the X-TAXII-Date-Added-First/Last headers are populated

### Related issues

* #2928

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

This needs discussion on fixing how the /manifest endpoint will end up needing 2 queries without making further changes.

### CODE REVIEW

**Comment by rlynch-ironnet** [comment]
Leaving this as a WIP until we figure out if domain/taxii.js should be updated, or if this is a good enough fix for now.

File: opencti-platform/opencti-graphql/src/http/httpTaxii.js [hunk 1]
```diff
@@ -45,6 +45,9 @@ const rebuildParamsForObject = (id, req) => {
   const argsMatch = { id, spec_version, version };
   return { added_after, limit, next, match: argsMatch };
 };
+const getUpdatedAt = (obj) => {
+  return obj?.extensions?.['extension-definition--ea279b3e-5c71-4632-ac08-831c66a786ba']?.updated_at;
```
Comment (richard-julien) [inline opencti-platform/opencti-graphql/src/http/httpTaxii.js]: Can you use the constant STIX_EXT_OCTI ? Thanks

**Review by richard-julien — CHANGES_REQUESTED** [review]

**Comment by rlynch-ironnet** [comment]
@richard-julien regarding `/manifest` doing a double query, should this PR refactor the functions to handle this, or deal with it in a later release?

**Comment by richard-julien** [comment]
> @richard-julien regarding `/manifest` doing a double query, should this PR refactor the functions to handle this, or deal with it in a later release?

Can you explain what you mean by `/manifest doing a double query`

Thanks

**Comment by rlynch-ironnet** [comment]
Previously, to get the manifest list, `/manifest` would call `const manifest = await restCollectionManifest(context, user, id, req.query);` which does a backend search. Each manifest object only has the dateAdded, not any dateModified, so we need to pull the stix objects as well, which performs another search via `const stix = await restCollectionStix(context, user, id, req.query);`

A better fix, would modify restCollectionManifest/Stix to accept a list of search objects, so only one search is performed instead.

**Comment by richard-julien** [comment]
> Previously, to get the manifest list, `/manifest` would call `const manifest = await restCollectionManifest(context, user, id, req.query);` which does a backend search. Each manifest object only has the dateAdded, not any dateModified, so we need to pull the stix objects as well, which performs another search via `const stix = await restCollectionStix(context, user, id, req.query);`
> 
> A better fix, would modify restCollectionManifest/Stix to accept a list of search objects, so only one search is performed instead.

Oh yes i miss read the file.

I think you can use directly data fetch from the manifest
```

  return {
    id: data.standard_id,
    date_added: data.created_at,
    version: data.updated_at,
    media_type: STIX_MEDIA_TYPE,
  };
```

The updated_at information is contains in version field

**Comment by rlynch-ironnet** [comment]
That should do the trick then.

File: opencti-platform/opencti-graphql/src/http/httpTaxii.js [hunk 2]
```diff
@@ -156,9 +160,10 @@ const initTaxiiApi = (app) => {
       const args = rebuildParamsForObject(object_id, req);
       const stix = await restCollectionStix(context, user, id, args);
       const data = R.head(stix.objects);
-      res.set('X-TAXII-Date-Added-First', data?.updated_at);
-      res.set('X-TAXII-Date-Added-Last', data?.updated_at);
-      const versions = data ? [data.updated_at] : [];
+      const updatedAt = getUpdatedAt(data);
+      res.set('X-TAXII-Date-Added-First', updatedAt);
+      res.set('X-TAXII-Date-Added-Last', updatedAt);
```
Comment (richard-julien) [inline opencti-platform/opencti-graphql/src/http/httpTaxii.js:165]: Already problematic before your modification, but this should not be taken for last element?

Comment (rlynch-ironnet) [reply opencti-platform/opencti-graphql/src/http/httpTaxii.js:165]: In theory, we'd want to do something like this
```
      const versions = stix.objects.map(o => getUpdatedAt(o));
      res.set('X-TAXII-Date-Added-First', R.head(versions));
      res.set('X-TAXII-Date-Added-Last', R.last(versions));
      res.json({ versions });
```
That is, if OpenCTI supported versioned objects.

**Review by richard-julien — APPROVED** [review]

**Pull request merged**